src/bvals/comms/build_boundary_buffers.cpp

```diff
@@ -110,14 +111,17 @@ void BuildBoundaryBufferSubset(std::shared_ptr<MeshData<Real>> &md,
     tag = pmesh->tag_map.GetTag(pmb, nb);
     auto comm_label = v->label();
     mpi_comm_t comm = pmesh->GetMPIComm(comm_label);
+
 #else
       // Setting to zero is fine here since this doesn't actually get used when everything
       // is on the same rank
       mpi_comm_t comm = 0;
 #endif
 
     bool use_sparse_buffers = v->IsSet(Metadata::Sparse);
-    auto get_resource_method = [pmesh, buf_size]() {
+    auto get_resource_method = [pmesh, buf_size](int size) {
+      PARTHENON_REQUIRE(size <= buf_size,
+                        "Asking for a buffer that is larger than size of pool.");
```

Comment:
Is this expected to ever show up in production? If not should this be a `PARTHENON_DEBUG_REQUIRE`?

Reply:
:+1: